Program.cs

```diff
@@ -204,5 +182,36 @@ public Task Log(LogMessage msg)
         {
             return Task.Run(() => Console.WriteLine(msg.ToString()));
         }
+
+        public string[] BuildHelpLines()
+        {
+            List<string> arrayList = new List<string>();
+
+            Assembly asm = Assembly.GetExecutingAssembly(); // Get assembly
+
+            var results = from type in asm.GetTypes()
+                          where typeof(ModuleBase).IsAssignableFrom(type)
+                          select type; // Grab all types that inherit ModuleBase
+
+            foreach (Type t in results) // For each type in results
+            {
+                /* Grab MethodInfo of the type where the method has the attribute SummaryAttribute */
+                MethodInfo info = t.GetMethods().Where(x => x.GetCustomAttributes(typeof(SummaryAttribute), false).Length > 0).First();
+
+                /* Grab summary attribute */
+                SummaryAttribute summary = info.GetCustomAttribute(typeof(SummaryAttribute)) as SummaryAttribute;
+
+                /* Grab command attribute */
+                CommandAttribute command = info.GetCustomAttribute(typeof(CommandAttribute)) as CommandAttribute;
+
+                /* Both objects are non null, valid, so lets grab the attribute text */
+                if (summary != null && command != null)
+                {
+                    arrayList.Add("!" + command.Text + " - " + summary.Text);
```

Comment:
Just realized we need to block disabled commands from appearing in the help menu.

Putting this note here for later, so I don’t forget.